common/archiver/README.md

```diff
@@ -17,32 +18,59 @@ This README explains how to create new implementations for the History and Visib
 
-3. Implement the Archiver interface. Both Archiver interfaces contain three methods:
-  
+**Step 2: Implement the HistoryArchiver interface**
+
 ```go
-func Archive(ctx context.Context, URI string, request *ArchiveRequest, opts ...ArchiveOption) error
+type HistoryArchiver interface {
+	// Archive is used to archive a workflow history. When the context expires the method should stop trying to archive.
+	// Implementors are free to archive however they want, including implementing retries of sub-operations. The URI defines
+	// the resource that histories should be archived into. The implementor gets to determine how to interpret the URI.
+	// The Archive method may or may not be automatically retried by the caller. The ArchiveOptions are used
+	// to interact with these retries including giving the implementor the ability to cancel retries and record progress
+	// between retry attempts. 
+    Archive(context.Context, URI, *ArchiveHistoryRequest, ...ArchiveOption) error
+    
+    // Get is used to access an archived history. When context expires method should stop trying to fetch history.
+    // The URI identifies the resource from which history should be accessed and it is up to the implementor to interpret this URI.
+    Get(context.Context, URI, *GetHistoryRequest) (*GetHistoryResponse, error)
+    
+    // ValidateURI is used to define what a valid URI for an implementation is.
+    ValidateURI(URI) error
+}
+```
 
-func Get(ctx context.Context, URI string, request *GetRequest) (*GetResponse, error)
+**Step 3: Implement the VisibilityArchiver interface**
 
-func ValidateURI(URI string) error
+```go
+type VisibilityArchiver interface {
+    // ValidateURI is used to define what a valid URI for an implementation is.
+    ValidateURI(URI) error
+}
 ```
+VisibilityArchiver is plumbed throughout the codebase but the full visibility feature is not yet implemented.
+This means even once the VisibilityArchiver is implemented archival of visibility events will not occur. 
+Despite this implementors are still expected to implement this one method interface.
+
+**Step 4: Update provider to provide access to your implementation**
 
-Each archiver implementation should define a format of URI, which is used to describe the type of implementation and the target location of the `Archive()` or `Get()` operation. For example, the URI for filestore uses the format: `file:///path/to/dir`. The `ValidateURI()` method is responsible for checking if a given string is a valid URI.
+Modify the `./provider/provider.go` file so that the `ArchiverProvider` knows how to create an instance of your archiver. 
+Also, add configs for you archiver to static yaml config files and modify the `HistoryArchiverConfig` 
+and `VisibilityArchiverConfig` struct in the `../common/service/config.go` accordingly.
```

Comment:
I think the structs are named `HistoryArchiverProvider` and `VisibilityArchiverProvider` now.